File: mobile/lib/services/auth_service.dart
import '../data/repositories/auth_repository.dart';

class AuthService {
  final AuthRepository _authRepository = AuthRepository();

  Future<Map<String, dynamic>>login(String email, String password) async {
    return await _authRepository.login(email, password);
  }

  Future<void> register(String name, String email, String password, String phone) async {
    await _authRepository.register(name, email, password, phone);
  }
}

File: frontend/src/services/authService.ts
import api from './api'

export interface AuthUser {
  id: number
  name: string
  email: string
  phone?: string
}

export interface Credentials {
  email: string
  password: string
}

export interface RegisterForm extends Credentials {
  name: string
  phone: string
}

interface AuthPayload {
  data: {
    token: string
    user: AuthUser
  }
}

export const authService = {
  async login(credentials: Credentials) {
    const response = await api.post<AuthPayload>('/auth/login', credentials)
    return response.data
  },

  async register(form: RegisterForm) {
    const response = await api.post<AuthPayload>('/auth/register', form)
    return response.data
  },
}
